client/src/reader/components/section/Text.js

```diff
@@ -104,6 +104,7 @@ export default class Text extends Component {
   // filteredAnnotations), but is in the unfiltered collection, then we can remove
   // it from the URL because the user has filtered it off the screen for now.
   checkRequestAnnotationHash() {
+    const { location } = this.props;
```

Comment:
This change requires some testing. Are you sure that this.props.location always === window.location?

Reply:
I double checked some documentation and it seems like it does, (especially with react-router) but I'll go with `const location = window.location` to be safe and clear.